UKHO.PeriodicOutputService/UKHO.BESS.ConfigurationService/Services/ConfigurationService.cs

```diff
@@ -213,13 +226,30 @@ public bool CheckConfigFrequencyAndSaveQueueDetails(IList<BessConfig> bessConfig
 
                         int? totalFileSize = encCells.Select(i => i.Item2).Sum();
 
-                        /* -- save details to message queue --
-                         *
-                         *
-                         *
-                         */
+                        double fileSizeInMb = CommonHelper.ConvertBytesToMegabytes(totalFileSize!.Value);
+
+                        int BESSize = Convert.ToInt16(configuration["BESSizeInMB"]);
+
+                        if (fileSizeInMb > BESSize)
+                        {
+                            logger.LogWarning(EventIds.BessSizeExceedsThreshold.ToEventId(), "Bespoke Exchange Set size {fileSizeInMb}MB which is more than the threshold :{BESSize}MB, Bespoke Exchange Set will not be created for file:{FileName} | _X-Correlation-ID : {CorrelationId}", Math.Round(fileSizeInMb, 2), BESSize, config.FileName, CommonHelper.CorrelationID);
+
+                            continue;
+                        }
+
+                        //--save details to message queue --
+                        IEnumerable<string> encCellNames = encCells.Select(i => i.Item1).ToList();
+
+                        Task<bool> success = azureBlobStorageService.SetConfigQueueMessageModelAndAddToQueue(config, encCellNames, totalFileSize);
```

Comment:
This asynchronous method is not awaited/waited, should it be?

Reply:
resolved.